### DISCUSSION TITLE
Train top-down id model

### DISCUSSION THREAD

**Discussion by Young-eng** [talmolab/sleap#1809] [Help!] [answered]
I want to evaluate tracking algorithms and get the metrics by training my custom datasets. I have trained top-down model with my datasets following the instructions in Tutorial, but when I try to train top down id model with my datasets, there are some errors during second stage : `sleap-train multi_class_topdown.json 2024_0612_fish_512_train_labels.v001.pkg.slp`

Error like this:
![image](https://github.com/talmolab/sleap/assets/55878284/5f9f548c-3e3c-458c-9a5c-42c667274993)

When I try to train top-down ID model with the example dataset: `drosophila-melanogaster-courtship`，I meet a different error during second stage:

![image](https://github.com/talmolab/sleap/assets/55878284/82668441-9788-486f-8404-10202782f188)

There are no errors when training top-down model with these datasets. So I am not very clear about the training processing of top-down ID model, and do not find similiar problems in Disscusion. Are there any other methods that I can evaluate tracking algorithms in Sleap on my custom datasets? Thanks for your help!

**Comment by gitttt-1234** [comment, marked as answer]
Since this was posted, we've moved all training and inference to [**sleap-nn**](https://nn.sleap.ai/latest/), which fully supports the `multi_class_topdown` model type and fixes the errors you encountered. There's a [sample config](https://nn.sleap.ai/latest/configuration/samples/) for top-down ID training you can use as a starting point. (**NOTE**: for training ID models, all instances in your training data must have consistent track labels assigned). At the moment, `sleap-nn eval` currently evaluates pose estimation quality (OKS, distance errors, PCK) but doesn't yet have tracking-specific metrics (MOTA, ID switches, etc.). 

Closing this in favor of the updated sleap-nn pipeline, feel free to open a new discussion if you run into anything or have any questions!
